spec/models/user_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is invalid without a name' do
+    user = User.new(name: nil)
+    expect(user).to_not be_valid
+    expect(user.errors[:name]).to include("can't be blank")
+  end
+
+  it 'is invalid when posts counter is not an integer' do
+    user = User.new(name: 'John Isac', posts_counter: 'abc')
+    expect(user).to_not be_valid
+    expect(user.errors[:posts_counter]).to include('is not a number')
+  end
+
+  it 'is invalid when posts counter is less than zero' do
+    user = User.new(name: 'John Isac', posts_counter: -1)
+    expect(user.valid?).to be false
+    expect(user.errors[:posts_counter]).to include('must be greater than or equal to 0')
+  end
+end
```

Comment:
- [ ] Great job adding the validations and testing them 👏. However, as stated in the requirement, all your models' methods should be tested too. So please, I would suggest you test the `recent_posts` method.

### Requirement :point_down: 
![Screenshot from 2023-12-15 16-53-41](https://github.com/Surafels/Blog-app/assets/89228743/2e79dda6-b073-458d-8fd9-45bc19a9024c)